roles/rsyslog/tasks/main.yml

```diff
@@ -76,16 +76,18 @@
             mode: '0700'
           changed_when: false
 
-        - name: Archive the contents of {{ __rsyslog_config_dir }} to
-            the backup dir
-          vars:
-            __now: "{{ lookup('pipe', 'date +%s') }}"
+        - name: Get backup file stat
+          stat:
+            path: "{{ __rsyslog_backup_dir }}/backup.tgz"
+          register: __backup_stat
+
+        - name: Archive the original contents of {{ __rsyslog_config_dir }}
+            to the backup dir
           command: >
-            tar -cvzPf "{{ __rsyslog_backup_dir }}/backup{{ __now }}.tgz"
-                /etc/rsyslog.conf "{{ __rsyslog_config_dir }}"
-          args:
-            warn: false  # suppress 'unarchive' warning
+            tar -cvzPf "{{ __rsyslog_backup_dir }}/backup.tgz"
+              /etc/rsyslog.conf "{{ __rsyslog_config_dir }}"
           changed_when: false
```

Comment:
do you still need the `changed_when` here?

Reply:
If we consider creating a backup file is a `change`, we don't need it. And without the statement, it doesn't break the idempotence. So, right. it's not needed any more. Removing it. Thanks!